pkg/claims/result.go

```diff
@@ -66,3 +69,21 @@ func (r Result) NewOutput(name string, data []byte) Output {
 		Value:         data,
 	}
 }
+
+func (r Result) FilterSensitiveOutputs(result driver.OperationResult, bun bundle.Bundle, store secrets.Store) []Output {
+	o := make([]Output, 0, len(result.Outputs))
+	for name, value := range result.Outputs {
+		sensitive, err := bun.IsOutputSensitive(name)
+		ot := r.NewOutput(name, []byte(value))
+		if err != nil || !sensitive {
+			o = append(o, ot)
+			continue
+		}
+
+		ot = ot.FormatSensitive()
```

Comment:
It would be really easy to accidentally call FormatSensitive function before you saved the original value of the output (before overwriting the value with the secret key).

I would prefer that we don't have Output.Value mean two things depending on the context: either it's the actual output value or it's a key.

It seems that you are persisting the output with the value set to the secret key, correct? How about instead we added a field to Output, maybe `Output.SecretKey` with the location of the secret, so that we don't use Value for two different purposes. Then change the function name to something like "FlagAsSecret", so it's more clear what it is doing.